addons/account_asset/account_asset.py

```diff
@@ -1,351 +1,325 @@
-                if asset.prorata:
-                    amount = amount_to_depr / asset.method_number
+                if self.prorata and self.category_id.type == 'purchase':
+                    amount = amount_to_depr / self.method_number
                     days = total_days - float(depreciation_date.strftime('%j'))
-                    if i == 1:
-                        amount = (amount_to_depr / asset.method_number) / total_days * days
-                    elif i == undone_dotation_number:
-                        amount = (amount_to_depr / asset.method_number) / total_days * (total_days - days)
-            elif asset.method == 'degressive':
-                amount = residual_amount * asset.method_progress_factor
-                if asset.prorata:
+                    if sequence == 1:
+                        amount = (amount_to_depr / self.method_number) / total_days * days
+                    elif sequence == undone_dotation_number:
+                        amount = (amount_to_depr / self.method_number) / total_days * (total_days - days)
+            elif self.method == 'degressive':
+                amount = residual_amount * self.method_progress_factor
+                if self.prorata:
                     days = total_days - float(depreciation_date.strftime('%j'))
-                    if i == 1:
-                        amount = (residual_amount * asset.method_progress_factor) / total_days * days
-                    elif i == undone_dotation_number:
-                        amount = (residual_amount * asset.method_progress_factor) / total_days * (total_days - days)
+                    if sequence == 1:
+                        amount = (residual_amount * self.method_progress_factor) / total_days * days
+                    elif sequence == undone_dotation_number:
+                        amount = (residual_amount * self.method_progress_factor) / total_days * (total_days - days)
         return amount
 
-    def _compute_board_undone_dotation_nb(self, cr, uid, asset, depreciation_date, total_days, context=None):
-        undone_dotation_number = asset.method_number
-        if asset.method_time == 'end':
-            end_date = datetime.strptime(asset.method_end, '%Y-%m-%d')
+    @api.multi
+    def _compute_board_undone_dotation_nb(self, depreciation_date, total_days):
+        undone_dotation_number = self.method_number
+        if self.method_time == 'end':
+            end_date = datetime.strptime(self.method_end, '%Y-%m-%d')
             undone_dotation_number = 0
             while depreciation_date <= end_date:
-                depreciation_date = (datetime(depreciation_date.year, depreciation_date.month, depreciation_date.day) + relativedelta(months=+asset.method_period))
+                depreciation_date = (datetime(depreciation_date.year, depreciation_date.month, depreciation_date.day) + relativedelta(months=+self.method_period))
```

Comment:
are you sure you need Datetime objects for depreciation_date and end_date? aren't Date enough?